react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      password: ''
+    };
+
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({
+      username: event.target.value
+    });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({
+      password: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit} >
+        <label htmlFor="username">Username</label>
```

Comment:
The `htmlFor` prop should match the `id` prop of the associated `input`. So if you are using `htmlFor` on the label the `input` should have a matching `id`.

Reply:
@scbowler got it! I thought it was supposed to match the `name` attribute of the associated input, but that makes sense! Thank you!